src-cljs/atom_parinfer/core.cljs

```diff
@@ -273,6 +273,42 @@
 
 (add-watch *editor-states :toggle-status-classes toggle-status-classes!)
 
+;;------------------------------------------------------------------------------
+;; Error Markers
+;;------------------------------------------------------------------------------
+
+(def error-marker-class "parinfer-error-marker")
+(def error-marker-ids (atom #{}))
+
+(defn add-error-marker
+  [js-editor start-row js-error]
+  (let [row (+ (oget js-error "lineNo") start-row)
+        col (oget js-error "x")
+        range (array (array row col)
+                     (array row (inc col)))
+        marker (ocall js-editor "markBufferRange" range
+                 (js-obj "invalidate" "never"))
+        decorator (ocall js-editor "decorateMarker" marker
+                    (js-obj "type" "highlight"
+                            "class" error-marker-class))]
+    (swap! error-marker-ids conj (oget marker "id"))))
+
+(defn marker-inside?
+  [marker start-row end-row]
+  (let [marker-row (oget (ocall marker "getBufferRange") "start" "row")]
+    (<= start-row marker-row end-row)))
+
+(defn get-error-markers
+  [js-editor start-row end-row]
+  (->> (ocall js-editor "findMarkers")
+       (filter #(@error-marker-ids (oget % "id")))
+       (filter #(marker-inside? % start-row end-row))))
+
+(defn clear-error-markers
+  [js-editor start-row end-row js-error]
+  (doseq [marker (get-error-markers js-editor start-row end-row)]
+    (swap! error-marker-ids disj (oget marker "id"))
```

Comment:
Would prefer to not see `swap!` inside a `doseq`

Can you make `swap!` a single operation here?